rollup/da_syncer/da/commitV0.go

```diff
@@ -129,28 +130,36 @@ func (c *CommitBatchDAV0) Blocks() []*PartialBlock {
 	return blocks
 }
 
-func getTotalMessagesPoppedFromChunks(decodedChunks []*codecv0.DAChunkRawTx) int {
+func getTotalMessagesPoppedFromChunks(decodedChunks []*encoding.DAChunkRawTx) int {
 	totalL1MessagePopped := 0
 	for _, chunk := range decodedChunks {
 		for _, block := range chunk.Blocks {
-			totalL1MessagePopped += int(block.NumL1Messages)
+			totalL1MessagePopped += int(block.NumL1Messages())
 		}
 	}
 	return totalL1MessagePopped
 }
 
 func getL1Messages(db ethdb.Database, parentTotalL1MessagePopped uint64, skippedBitmap []byte, totalL1MessagePopped int) ([]*types.L1MessageTx, error) {
 	var txs []*types.L1MessageTx
-
 	decodedSkippedBitmap, err := encoding.DecodeBitmap(skippedBitmap, totalL1MessagePopped)
 	if err != nil {
 		return nil, fmt.Errorf("failed to decode skipped message bitmap: err: %w", err)
 	}
 
+	isL1MessageSkipped := func(skippedBitmap []*big.Int, index uint64) bool {
+		if index >= uint64(len(skippedBitmap))*256 {
+			return false
+		}
+		quo := index / 256
+		rem := index % 256
+		return skippedBitmap[quo].Bit(int(rem)) == 1
+	}
+
 	// get all necessary l1 messages without skipped
 	currentIndex := parentTotalL1MessagePopped
 	for index := 0; index < totalL1MessagePopped; index++ {
-		if encoding.IsL1MessageSkipped(decodedSkippedBitmap, currentIndex-parentTotalL1MessagePopped) {
+		if isL1MessageSkipped(decodedSkippedBitmap, currentIndex-parentTotalL1MessagePopped) {
```

Comment:
@NazariiDenha do you think we should add this function in da-codec repo?

Reply:
Previously it was there and I think it was better

Reply:
OK.

Reply:
done in [bbd7b73](https://github.com/scroll-tech/go-ethereum/pull/1068/commits/bbd7b73b9bdbe4eb500aed604002befc8d16664c).